sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/randomExpressions.scala

```diff
@@ -50,22 +54,27 @@ abstract class RDG extends LeafExpression with Nondeterministic {
 
   override def dataType: DataType = DoubleType
 
-  // NOTE: Even if the user doesn't provide a seed, Spark SQL adds a default seed.
-  override def sql: String = s"$prettyName($seed)"
+  override def inputTypes: Seq[AbstractDataType] = Seq(TypeCollection(IntegerType, LongType))
 }
 
 /** Generate a random column with i.i.d. uniformly distributed values in [0, 1). */
+// scalastyle:off line.size.limit
 @ExpressionDescription(
-  usage = "_FUNC_(a) - Returns a random column with i.i.d. uniformly distributed values in [0, 1).")
-case class Rand(seed: Long) extends RDG {
-  override protected def evalInternal(input: InternalRow): Double = rng.nextDouble()
+  usage =
+    """
+      _FUNC_() - Returns a random column with i.i.d. uniformly distributed values in [0, 1].
+        seed is given randomly.
+
+      _FUNC_(seed) - Returns a random column with i.i.d. uniformly distributed values in [0, 1].
+        seed should be an integer/long/NULL literal.
+    """,
+  extended = "> SELECT _FUNC_();\n 0.9629742951434543\n> SELECT _FUNC_(0);\n 0.8446490682263027\n> SELECT _FUNC_(NULL);\n 0.8446490682263027")
```

Comment:
Given the discussion in the other PR, should the 'usage' message be shorter and push some of this into 'extended'? whatever would be consistent.